jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/bpmn2/BPMN2DataServiceImpl.java

```diff
@@ -47,17 +60,32 @@
 public class BPMN2DataServiceImpl implements DefinitionService, DeploymentEventListener {
     
     private static final Logger logger = LoggerFactory.getLogger(BPMN2DataServiceImpl.class);
+    
     private static final BPMN2DataServiceSemanticModule MODULE = new BPMN2DataServiceSemanticModule();
+    private static final BPMN2DataServiceExtensionSemanticModule EXTENSIONS_MODULE = new BPMN2DataServiceExtensionSemanticModule(MODULE);
     
-    private ConcurrentHashMap<String, Map<String, ProcessDescRepoHelper>> definitionCache = 
+    private ConcurrentMap<String, Map<String, ProcessDescRepoHelper>> definitionCache = 
     		new ConcurrentHashMap<String, Map<String, ProcessDescRepoHelper>>();
+   
+    private static final String [] dialectNames = { "java", "mvel" };
+    private static final ProcessDialect [] dataServiceDialects = { 
+            new DataServiceJavaProcessDialect(),
+            new DataServiceMvelProcessDialect()
```

Comment:
Ah, yes. :) 

See the javadoc here: https://github.com/droolsjbpm/jbpm/pull/254/files#diff-6e6dc1ba9a9f5984b8dc659bd08ee925R17

Basically, there is a layered structure of ThreadLocal instances. The ThreadLocal in the data service dialects switches between the singleton instance of the (normal) ActionBuilder (and other builder) and the singleton instance of the data service action builder. Within the data service action builder, there's _another_ ThreadLocal that refers to the thread local process description repo helper instance..